internal/imgexpirer/imgexpirer.go

```diff
@@ -25,22 +27,30 @@ type Manager struct {
 	objectHandler s3wrapper.API
 	eventsHandler events.Handler
 	deleteTime    time.Duration
+	leaderElector leader.Leader
 }
 
-func NewManager(objectHandler s3wrapper.API, eventsHandler events.Handler, deleteTime time.Duration) *Manager {
+func NewManager(objectHandler s3wrapper.API, eventsHandler events.Handler, deleteTime time.Duration, leaderElector leader.ElectorInterface) *Manager {
 	return &Manager{
 		objectHandler: objectHandler,
 		eventsHandler: eventsHandler,
 		deleteTime:    deleteTime,
+		leaderElector: leaderElector,
 	}
 }
 
 func (m *Manager) ExpirationTask() {
+	if !m.leaderElector.IsLeader() {
+		return
+	}
 	ctx := requestid.ToContext(context.Background(), requestid.NewID())
 	m.objectHandler.ExpireObjects(ctx, imagePrefix, m.deleteTime, m.DeletedImageCallback)
 }
 
 func (m *Manager) DeletedImageCallback(ctx context.Context, log logrus.FieldLogger, objectName string) {
+	if !m.leaderElector.IsLeader() {
```

Comment:
adding it here. There is no big issue to go threw all object but i don't want to delete object if pod is not a leader